torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -281,6 +283,21 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                     node.replace_all_uses_with(guard_scalar(val))
                     graph.erase_node(node)
 
+    should_restart = False
+    for i, node in enumerate(graph.nodes):
+        if node.op == "placeholder" and len(node.users) == 0:
```

Comment:
If we want to optimize for performance perhaps it make sense to break early once we iterate past the placeholders.